local/local.go

```diff
@@ -247,3 +263,33 @@ func unparseFlag(flags *pflag.FlagSet, flag *pflag.Flag) []string {
 	}
 	return result
 }
+
+type buildAgentSettings struct {
+	LatestSha256 string
+}
+
+func loadBuildAgentShaFromConfig() (string, error) {
+	if _, err := os.Stat(buildAgentSettingsPath()); os.IsNotExist(err) {
+		// Settings file does not exist.
+		return "", nil
+	}
+
+	file, err := os.Open(buildAgentSettingsPath())
+	if err != nil {
+		return "", errors.Wrap(err, "Could not open build settings config")
+	}
+	defer file.Close()
+
+	var settings buildAgentSettings
+
+	if err := json.NewDecoder(file).Decode(&settings); err != nil {
```

Comment:
I would guess that the use of `NewDecoder` here is mostly to make use of the fact that it takes a `Reader`, making it easier to use it with `File`, right?

I would instinctively have read the file into memory and used the singleton decoder from `encoding/json`.

```go
buf, err := io.ReadAll(file)
if err != nil {
  return "", errors.Wrap(err, "couldn't read from build settings file")
}

if err = json.Unmarshal(buf, &settings); err != nil {
```

I feel like allocating a decoder with a reference to a buffer is riskier memory management, but then again, calling `ReadAll` on a huge file would also be a problem. 🤔 

----

No blocker here, just opening the conversation.

Reply:
Marginal risk here, as it is running on a users machine with a file the user has to create

Reply:
You're right, both methods have trade-offs and imo your proposition is better for little files so yes I can go for it

Reply:
I agree, @danmux. Especially since CLIs have very short runs, so no risk of accreting memory leaks.

Really just chatting and getting your feedback on this. 😁